staging/src/k8s.io/component-base/metrics/prometheus/feature/metrics.go

```diff
@@ -34,9 +38,6 @@ var (
 		},
 		[]string{"name", "stage"},
 	)
-)
-
-func init() {
```

Comment:
Thats right, `--allow-metric-labels` isnt honored anywhere except for scheduler metrics (support recently added in https://github.com/kubernetes/kubernetes/pull/126762)

Ref: https://github.com/kubernetes/kubernetes/issues/126526#issuecomment-2350516080 which does say that we need to fix the issue in all component metrics.

Reply:
is there a more systemic way we can fix this? individually adjusting all init paths seems like it will immediately regress as more metrics are added in various places

Reply:
yeah this was solved by https://github.com/kubernetes/kubernetes/pull/128100 in a much better way, thanks to @yongruilin 
I'll close this PR since its not needed now.